pallets/manta-pay/src/types.rs

```diff
@@ -86,72 +131,96 @@ pub type Proof = [u8; PROOF_LENGTH];
 }
 
 impl Asset {
     /// Builds a new [`Asset`] from `id` and `value`.
     #[inline]
-    pub fn new(id: StandardAssetId, value: Balance) -> Self {
+    pub fn new(id: AssetId, value: AssetValue) -> Self {
         Self { id, value }
     }
 }
 
-/// Encrypted Note
+impl From<config::Asset> for Asset {
+    #[inline]
+    fn from(asset: config::Asset) -> Self {
+        Self {
+            id: encode(asset.id),
+            value: asset.value,
+        }
+    }
+}
+
+impl TryFrom<Asset> for config::Asset {
+    type Error = Error;
+
+    #[inline]
+    fn try_from(asset: Asset) -> Result<Self, Self::Error> {
+        Ok(Self {
+            id: decode(asset.id)?,
+            value: asset.value,
+        })
+    }
+}
+
+/// Outgoing Ciphertext
+pub type OutgoingCiphertext = [[u8; 32]; 2];
+
+/// Outgoing Note
 #[cfg_attr(
     feature = "rpc",
     derive(Deserialize, Serialize),
     serde(crate = "manta_util::serde", deny_unknown_fields)
 )]
-#[derive(Clone, Debug, Decode, Encode, Eq, Hash, MaxEncodedLen, PartialEq, TypeInfo)]
-pub struct EncryptedNote {
+#[derive(Clone, Debug, Decode, Default, Encode, Eq, Hash, MaxEncodedLen, PartialEq, TypeInfo)]
+pub struct OutgoingNote {
     /// Ephemeral Public Key
     pub ephemeral_public_key: Group,
 
-    /// Ciphertext
-    #[cfg_attr(
-        feature = "rpc",
-        serde(
-            with = "manta_util::serde_with::As::<[manta_util::serde_with::Same; CIPHER_TEXT_LENGTH]>"
-        )
-    )]
-    pub ciphertext: Ciphertext,
-}
+    /// Tag
+    pub tag: Tag,
 
-impl Default for EncryptedNote {
-    #[inline]
-    fn default() -> Self {
-        Self {
-            ephemeral_public_key: [0; EPHEMERAL_PUBLIC_KEY_LENGTH],
-            ciphertext: [0; CIPHER_TEXT_LENGTH],
-        }
-    }
+    /// Ciphertext
+    pub ciphertext: OutgoingCiphertext,
 }
 
-impl From<config::EncryptedNote> for EncryptedNote {
+impl From<v2::OutgoingNote> for OutgoingNote {
     #[inline]
-    fn from(encrypted_note: config::EncryptedNote) -> Self {
-        let encrypted_note = encrypted_note.ciphertext;
+    fn from(note: v2::OutgoingNote) -> Self {
         Self {
-            ephemeral_public_key: encode(encrypted_note.ephemeral_public_key),
-            ciphertext: encrypted_note.ciphertext.into(),
+            ephemeral_public_key: encode(note.ciphertext.ephemeral_public_key),
+            tag: encode(note.ciphertext.ciphertext.tag.0),
+            ciphertext: Array::from_iter(
+                note.ciphertext.ciphertext.message[0].0.iter().map(encode),
+            )
+            .into(),
         }
     }
 }
 
-impl TryFrom<EncryptedNote> for config::EncryptedNote {
+impl TryFrom<OutgoingNote> for v2::OutgoingNote {
     type Error = Error;
 
     #[inline]
-    fn try_from(encrypted_note: EncryptedNote) -> Result<Self, Self::Error> {
+    fn try_from(note: OutgoingNote) -> Result<Self, Self::Error> {
         Ok(Self {
-            header: (),
+            header: EmptyHeader::default(),
```

Comment:
double-check defaults

Reply:
The default is PhantomData, so should be ok.

Reply:
basically in manta-crypto we define the Encryption trait for encryption schemes
the Encryption trait is a supertrait of HeaderType
but OutgoingNote doesn't need a header, so we just give it EmptyHeader because HeaderType still needs to be implemented